tests/unit/test_util.py

```diff
@@ -0,0 +1,22 @@
+import unittest
+
+from src.utils import valid_address, utc_now
+
+
+class TestUtils(unittest.TestCase):
```

Comment:
In my opinion, we should also be testing that `self._update_memory()` gets called only once if we call `get_affiliate_data` twice within the cache TTL.

Reply:
Oh yes. We will.